joblib/test/test_hashing.py

```diff
@@ -70,8 +72,8 @@ def f(self, x):
 
 class KlassWithCachedMethod(object):
 
-    def __init__(self):
-        mem = Memory(cachedir=test_memory_env['dir'])
+    def __init__(self, tmpdir_path):
```

Comment:
Nope, `tmpdir_path` used that way would take in the global reference - the function itself. Pytest actually executes these functions marked with `@fixture` before test execution and automatically provides them as function arguments whenever specified. So this `tmpdir_path` will be passed in constructor from inside a test method.